internal/konnect/node_agent_test.go

```diff
@@ -438,21 +445,22 @@ func TestNodeAgentUpdateNodes(t *testing.T) {
 
 			ctx := context.Background()
 			go func() {
-				err := nodeAgent.Start(ctx)
-				require.NoError(t, err)
+				require.NoError(t, nodeAgent.Start(ctx))
 			}()
 
 			if tc.configStatus != nil {
 				configStatusSubscriber.NotifyConfigStatus(ctx, *tc.configStatus)
 			}
 
 			gatewayClientsChangesNotifier.Notify()
+
+			var nodes []*konnect.NodeItem
+			require.Eventually(t, func() bool {
+				nodes = nodeService.dumpNodes()
+				return len(nodes) == tc.numNodes
+			}, timeout, tick, "number of nodes in RG should be %d", tc.numNodes)
+
 			require.Eventually(t, func() bool {
-				// check number of nodes in RG.
-				nodes := nodeService.dumpNodes()
-				if len(nodes) != tc.numNodes {
-					return false
-				}
```

Comment:
It might be useful to log all of the actual nodes in case any of the assertions in `Eventually` closure fails, just in case it happens again despite the extended timeout. 

I tried to understand why this test is flaky, but couldn't reproduce it locally (tried `go test -count=1000` a couple of times with no luck). There must be some subtle race condition :/

Reply:
👍 I've changed it to print the nodes but couldn't find a better solution than a for loop ( just printing it using asserts was printing a slice of pointer values). 

Feel free to change it if you know a better way.